aten/src/ATen/native/quantized/cpu/q_avgpool.cpp

```diff
@@ -12,212 +13,224 @@ namespace at {
 namespace native {
 namespace {
 
-inline int start_index(int a, int b, int c) {
-  return (int)std::floor((float)(a * c) / b);
-}
+template <typename scalar_t, typename underlying_t>
+static void avg_pool2d_out_frame(
+          scalar_t *input_data,
+          scalar_t *output_data,
+          int64_t nbatch,
+          int64_t nInputPlane,
+          int64_t inputWidth,
+          int64_t inputHeight,
+          int64_t outputWidth,
+          int64_t outputHeight,
+          int kW,
+          int kH,
+          int dW,
+          int dH,
+          int padW,
+          int padH,
+          bool count_include_pad,
+          c10::optional<int64_t> divisor_override)
+{
+  at::parallel_for(0, nInputPlane, 0, [&](int64_t start, int64_t end) {
+    for (auto k = start; k < end; k++)
+    {
+      int64_t p;
+      for(p = 0; p < nbatch; p++)
+      {
+        int64_t xx, yy;
+        /* For all output pixels... */
+        scalar_t *ptr_output = output_data + p*nInputPlane*outputWidth*outputHeight + k*outputWidth*outputHeight;
+        const scalar_t *ptr_input = input_data + p*nInputPlane*inputWidth*inputHeight + k*inputWidth*inputHeight;
+        int64_t i;
+        for(i = 0; i < outputWidth*outputHeight; i++)
+          (ptr_output+i)->val_ = 0;
+
+        for(yy = 0; yy < outputHeight; yy++)
+        {
+          for(xx = 0; xx < outputWidth; xx++)
+          {
+            /* Compute the mean of the input image... */
+            int64_t hstart = yy * dH - padH;
+            int64_t wstart = xx * dW - padW;
+            int64_t hend = std::min(hstart + kH, inputHeight + padH);
+            int64_t wend = std::min(wstart + kW, inputWidth + padW);
+            int pool_size = (hend - hstart) * (wend - wstart);
```

Comment:
Maybe we should use `int64_t` for `pool_size` and `divide_factor`? Or we should change all other variables to be `int`/`int32_t`? Just make it consistent with higher precision (`int64_t`) or the same as C2 (`int32_t`).